src/Mod/Part/App/WireJoiner.cpp

```diff
@@ -0,0 +1,2524 @@
+# include <BRep_Builder.hxx>
+# include <BRep_Tool.hxx>
+# include <BRepBndLib.hxx>
+# include <BRepBuilderAPI_Copy.hxx>
+# include <BRepBuilderAPI_MakeEdge.hxx>
+# include <BRepBuilderAPI_MakeWire.hxx>
+# include <BRepBuilderAPI_MakeFace.hxx>
+# include <BRepClass_FaceClassifier.hxx>
+# include <BRepExtrema_DistShapeShape.hxx>
+# include <BRepGProp.hxx>
+# include <BRepTools.hxx>
+# include <BRepTools_WireExplorer.hxx>
+# include <gp_Pln.hxx>
+# include <GeomAdaptor_Curve.hxx>
+# include <GeomLProp_CLProps.hxx>
+# include <GProp_GProps.hxx>
+# include <ShapeAnalysis_Wire.hxx>
+# include <ShapeFix_ShapeTolerance.hxx>
+# include <ShapeExtend_WireData.hxx>
+# include <ShapeFix_Wire.hxx>
+# include <ShapeFix_Shape.hxx>
+# include <TopExp.hxx>
+# include <TopExp_Explorer.hxx>
+# include <TopTools_HSequenceOfShape.hxx>
+#endif
+
+#include <BRepTools_History.hxx>
+#include <ShapeBuild_ReShape.hxx>
+
+#include <unordered_map>
+#include <unordered_set>
+#include <deque>
+#include <boost_geometry.hpp>
+
+#include <Base/Console.h>
+#include <Base/Exception.h>
+#include <Base/Tools.h>
+#include <Base/Sequencer.h>
+#include <Base/Parameter.h>
+#include <App/Application.h>
+
+#include "WireJoiner.h"
+
+#include "Geometry.h"
+#include "PartFeature.h"
+#include "TopoShapeOpCode.h"
+#include "TopoShapeMapper.h"
+
+namespace bg = boost::geometry;
+namespace bgi = boost::geometry::index;
+
+using RParameters = bgi::linear<16>;
+
+BOOST_GEOMETRY_REGISTER_POINT_3D_GET_SET(
+        gp_Pnt,double,bg::cs::cartesian,X,Y,Z,SetX,SetY,SetZ)
+
+FC_LOG_LEVEL_INIT("WireJoiner",true, true)
+
+using namespace Part;
+
+static inline void getEndPoints(const TopoDS_Edge &eForEndPoints, gp_Pnt &p1, gp_Pnt &p2) {
+    p1 = BRep_Tool::Pnt(TopExp::FirstVertex(eForEndPoints));
+    p2 = BRep_Tool::Pnt(TopExp::LastVertex(eForEndPoints));
+}
+
+static inline void getEndPoints(const TopoDS_Wire &wire, gp_Pnt &p1, gp_Pnt &p2) {
+    BRepTools_WireExplorer xp(wire);
+    p1 = BRep_Tool::Pnt(TopoDS::Vertex(xp.CurrentVertex()));
+    for(;xp.More();xp.Next()) {};
+    p2 = BRep_Tool::Pnt(TopoDS::Vertex(xp.CurrentVertex()));
+}
+
+static void _assertCheck(int line, bool cond, const char *msg)
+{
+    if (!cond) {
+        _FC_ERR(__FILE__, line, "Assert failed: " << msg);
+        throw Base::RuntimeError("Assertion failed");
+    }
+}
+
+#define assertCheck(cond)  _assertCheck(__LINE__, cond, #cond)
+
+class WireJoiner::WireJoinerP {
+public:
+    double myTol = Precision::Confusion();
+    double myTol2 = myTol * myTol;
+    double myAngularTol = Precision::Angular();
+    bool doSplitEdge = true;
+    bool doMergeEdge = true;
+    bool doOutline = false;
+    bool doTightBound = true;
+
+    std::string catchObject;
+    int catchIteration;
+    int iteration = 0;
+
+    using Box = bg::model::box<gp_Pnt>;
+
+    bool checkBBox(const Bnd_Box &box) const
+    {
+        if (box.IsVoid()) {
+            return false;
+        }
+        Standard_Real xMin, yMin, zMin, xMax, yMax, zMax;
```

Comment:
I see, for example in https://github.com/FreeCAD/FreeCAD/blob/0def330a9e2119a14bb349cc406c2aa365421ab4/src/3rdParty/salomesmesh/src/SMESH/GEOMUtils.cpp#L599, that this isn't the only place where `Standard_Real` variables are declared in the same line and not initialized.

Do we want to modify this one?

Reply:
Yes, since this file is new we should endeavor to fix as many linter warnings as possible. In this case, please do put each on its own line, and initialize them all.

Reply:
I would // NOLINT this one, because the six variables form a bounding box.  We aren't touching your reference example, which as 3rd party isn't even FC code,  but I would have moved val1 and val2 to separate lines and likely renamed them for that code.

Reply:
> I would // NOLINT this one

You mean what's in `WireJoiner.cpp` or what's in `GEOMUtils.cpp`?

By now I applied @chennes suggestion. I will revert the change if you want!